apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/HealthCheckStreamCaller.java

```diff
@@ -0,0 +1,36 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import com.nava.health.v1.HealthCheckRequest;
+import com.nava.health.v1.HealthCheckResponse;
+import com.nava.health.v1.HealthGrpc;
+import gov.cms.bfd.pipeline.rda.grpc.PreAdjudicatedClaim;
+import io.grpc.ManagedChannel;
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+/** GrpcStreamCaller implementation that calls the RDA HealthCheck service. */
+public class HealthCheckStreamCaller implements GrpcStreamCaller<HealthCheckResponse> {
+  private final String service;
+  private final HealthGrpc.HealthBlockingStub stub;
+
+  public HealthCheckStreamCaller(String service, ManagedChannel channel) {
+    this.service = service;
+    stub = HealthGrpc.newBlockingStub(channel);
+  }
+
+  @Override
+  public Iterator<HealthCheckResponse> callService(Duration maxRunTime) throws Exception {
+    final HealthCheckRequest request = HealthCheckRequest.newBuilder().setService(service).build();
+    return stub.withDeadlineAfter(maxRunTime.toMillis(), TimeUnit.MILLISECONDS).watch(request);
```

Comment:
It's not clear to me that this is the right deadline. I'd probably add a comment here detailing what this call does, with a focus on how long it might block for, what events it will fire (if any), and stuff like that.

This will be many folks' first encounter with a gRPC service (it is for me!) and you should take the opportunity to educate them a bit.